Lib/test/support/os_helper.py

```diff
@@ -716,3 +717,37 @@ def __exit__(self, *ignore_exc):
             else:
                 self._environ[k] = v
         os.environ = self._environ
+
+
+try:
+    import ctypes
+    kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+except (ImportError, AttributeError):
+    def subst_drive(path):
+        raise NotImplementedError('ctypes or kernel32 is not available')
+else:
+    ERROR_FILE_NOT_FOUND = 2
+    DDD_REMOVE_DEFINITION = 2
+    DDD_EXACT_MATCH_ON_REMOVE = 4
+    DDD_NO_BROADCAST_SYSTEM = 8
+
+    @contextlib.contextmanager
+    def subst_drive(path):
+        """Temporarily yield a substitute drive for a given path."""
+        for c in reversed(string.ascii_uppercase):
+            drive = f'{c}:'
+            if (not kernel32.QueryDosDeviceW(drive, None, 0) and
+                    ctypes.get_last_error() == ERROR_FILE_NOT_FOUND):
+                break
+        else:
+            raise FileExistsError('no available logical drive')
```

Comment:
Since this is only used for testing, maybe if there's no available drive the test should be skipped instead of failing:

```suggestion
            raise unittest.SkipTest('no available logical drive')
```